platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -3789,27 +3777,24 @@ void DisplayServerX11::_dispatch_input_event(const Ref<InputEvent> &p_event) {
 		if (windows.has(event_from_window->get_window_id())) {
 			Callable callable = windows[event_from_window->get_window_id()].input_event_callback;
 			if (callable.is_valid()) {
-				callable.callp((const Variant **)&evp, 1, ret, ce);
+				callable.call(p_event);
 			}
 		}
 	} else {
 		// Send to all windows.
 		for (KeyValue<WindowID, WindowData> &E : windows) {
 			Callable callable = E.value.input_event_callback;
 			if (callable.is_valid()) {
-				callable.callp((const Variant **)&evp, 1, ret, ce);
+				callable.call(p_event);
 			}
 		}
 	}
 }
 
 void DisplayServerX11::_send_window_event(const WindowData &wd, WindowEvent p_event) {
-	if (!wd.event_callback.is_null()) {
+	if (wd.event_callback.is_valid()) {
```

Comment:
I'm curious as to whether we have some rule about using `is_null()` and `is_valid()`. Long-term (Godot 5.0), maybe one should go away...

Reply:
I just realized this one is probably wrong, because `is_valid()` and `is_null()` are different with Callable 😬

Reply:
That's a good point. I often wonder how to model `if`s, in regards to making the positive branch the general or common case or either let it be the one that leads to a more easily readable expression as the condition.
That said, for some reason, I had failed to apply that mindset to expressions involving `is_valid/null()`. I acknowledge the readability they bring.